### ISSUE TITLE
Highcharts v12 with NextJS

### ISSUE THREAD

**Issue by julioxavierr** [highcharts/highcharts-react#502]
I'm using Highcharts with NextJS and when trying to update to Highcharts v12, I get the following error:

```
web:build:  ✓ Compiled successfully
   Collecting page data  .TypeError: Cannot read properties of undefined (reading 'createEvent')
web:build:     at j (/project/node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:2038)
web:build:     at setOptions (/project/node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:16987)
web:build:     at Object.compose (/project/node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts-more.js:9:23390)
```

From my research, it seems that this issue might be caused by trying to instance a chart on the server-side, which didn't seem to happen before.

While looking at the README of the project, I noticed that the following section was updated with a comment to only init modules if the Highcharts version is below v12.

https://github.com/highcharts/highcharts-react/blob/11c276dc53f9bca4e0750c4dacf4c880a98f7b73/README.md?plain=1#L423-L436

However, the NextJS section still says the module should be initiated:

https://github.com/highcharts/highcharts-react/blob/11c276dc53f9bca4e0750c4dacf4c880a98f7b73/README.md?plain=1#L140-L155

Which approach should be followed for NextJS? Could the server-side instance attempt be caused by something else?

**Comment by julioxavierr** [comment]
If I remove all the module inits, as the first piece of documentation says, it builds. However, it crashes when trying to render the chart:

```
Error: Highcharts error #17: www.highcharts.com/errors/17/?missingModuleFor=histogram
 - missingModuleFor: histogram
    at Object.eval (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:138)
    at j (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:2591)
    at T (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:61)
    at aR.initSeries (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:196458)
    at eval (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:212099)
    at Array.forEach (<anonymous>)
    at aR.firstRender (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:212077)
    at aR.eval (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:196375)
    at j (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:2591)
    at aR.init (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:195604)
    at new aR (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:195314)
    at Object.chart (webpack-internal:///../../node_modules/.pnpm/highcharts@12.0.1/node_modules/highcharts/highcharts.js:9:195188)
    at e (webpack-internal:///../../node_modules/.pnpm/highcharts-react-official@3.2.1_highcharts@12.0.1_react@18.3.1/node_modules/highcharts-react-official/dist/highcharts-react.min.js:1:2479)
    at eval (webpack-internal:///../../node_modules/.pnpm/highcharts-react-official@3.2.1_highcharts@12.0.1_react@18.3.1/node_modules/highcharts-react-official/dist/highcharts-react.min.js:1:3009)
    at commitHookEffectListMount (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:23184:26)
    at safelyCallCommitHookLayoutEffectListMount (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:22898:5)
    at reappearLayoutEffectsOnFiber (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:23554:13)
    at reappearLayoutEffects_complete (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:24872:7)
    at reappearLayoutEffects_begin (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:24860:7)
    at commitLayoutEffects_begin (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:24683:11)
    at commitLayoutEffects_begin (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:24690:11)
    at commitLayoutEffects (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:24646:3)
    at commitRootImpl (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26857:5)
    at commitRoot (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26716:5)
    at finishConcurrentRender (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26015:9)
    at performConcurrentWorkOnRoot (webpack-internal:///../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:25843:7)
    at workLoop (webpack-internal:///../../node_modules/.pnpm/scheduler@0.23.2/node_modules/scheduler/cjs/scheduler.development.js:266:34)
    at flushWork (webpack-internal:///../../node_modules/.pnpm/scheduler@0.23.2/node_modules/scheduler/cjs/scheduler.development.js:239:14)
    at MessagePort.performWorkUntilDeadline (webpack-internal:///../../node_modules/.pnpm/scheduler@0.23.2/node_modules/scheduler/cjs/scheduler.development.js:533:21)
```

**Comment by KamilKubik** [comment]
Hi [julioxavierr](https://github.com/julioxavierr), thanks for bringing this up! 

From the Highcharts `v12` you don't have to initialize the module - you should directly import it instead. Kindly refer to the [demo](https://stackblitz.com/edit/nextjs-kywyyf?file=app%2Fpage.tsx). Also, you have to run Highcharts in the client environment, not the server side (it was not supported in the previous versions either). 

I agree that the current documentation may not be clear on this. Therefore, let me know if there's still anything unclear, and I'll leave this thread open until the documentation is updated. 

Kind Regards!

**Comment by KamilKubik** [comment]
@julioxavierr thanks for sharing this example!

It may be worth mentioning the key difference between the `pages` and `app` routing approaches in Next.js (in terms of rendering Highcharts). When using the app routing approach _(introduced in Next.js v13)_ the `"use client"` directive is needed to render the route in the client environment, whereas using the pages routing, the route is rendered on the client by default. 

Regards!

**Comment by Bugrabugra** [comment]
@julioxavierr 
Can you achieve to run the code on development and prod environments? In my case when using react 19 with next 15, I can run on dev but I can not build. I will try your repos react and next versions.

**Comment by Bugrabugra** [comment]
no, I couldn't make it work in Next.js 15 or 14, with @julioxavierr 's demo versions or with latest version of Next.js. I have tried react-router@7 even. I am disabling SSR in every projects, there is no way it runs. I am importing map module like this

`import "highcharts/modules/map";`

or like

`
import HighchartsMap from "highcharts/modules/map";
HighchartsMap(Highcharts);
`

I have asked 50 questions to chatgpt as well, tried every suggestions. Even though it works on dev env, it doesn'T build the project, I get 

TypeError: Cannot read properties of undefined (reading 'Axis')

error every time. Any help would be great, I got a project from my company with a 2 weeks dead end and I got stucked at the beginning.

**Comment by Bugrabugra** [comment]
For anyone having the same problem with Next + Highcharts 12, I have tried every solution on internet and with chatgpt. Only solution was downgrading my highcharts to version 11. I have spent like 4 hours on this. here are the things I have tried

```
"use client";

import Highcharts from "highcharts";
import HighchartsReact from "highcharts-react-official";
import HighchartsMap from "highcharts/modules/map";
import HighchartsAccessibility from "highcharts/modules/accessibility";
import { useEffect, useState } from "react";


if (typeof Highcharts === "object") {
  HighchartsMap(Highcharts);
  HighchartsAccessibility(Highcharts);
}
```

or 

```"use client";

import Highcharts from "highcharts";
import HighchartsReact from "highcharts-react-official";
import "highcharts/modules/map";
import "highcharts/modules/accessibility";
import { useEffect, useState } from "react";```

none of them worked. By "none of them worked", I mean they run on dev but you get error on build if your next.config.ts is output: "export" (exports only js, css and html, no ssr).

for anyone, this is my working (on dev and build) lib versions

`    "highcharts": "^11.2.0",
    "highcharts-react-official": "^3.2.1",
    "next": "^15.0.4",
    "react": "^19.0.0",
    "react-dom": "^19.0.0",`

**Comment by julioxavierr** [comment]
Hey @Bugrabugra,

Have you tried loading isolating all your highcharts imports into a single component and loading it with `next/dynamic`?
That seems to have fixed the build error for me.

**Comment by KamilKubik** [comment]
Hi [Bugrabugra](https://github.com/Bugrabugra)!

It seems the issue comes from referencing the web API in the node environment during the build. Using the dynamic imports should work as @julioxavierr mentioned, but also, I'd like to refer you to this workaround (to make sure the modules load in the client environment):

```
"use client";

import { useEffect, useState } from "react";
import Highcharts from "highcharts";
import HighchartsReact from "highcharts-react-official";

const loadHighchartsModules = async (callback) => {
  Promise.all([
    import("highcharts/modules/map"),
    import("highcharts/modules/exporting"),
    import("highcharts/modules/accessibility"),
  ]).then(callback);
};

export default function Home() {
  const [options, setOptions] = useState(null);

  useEffect(() => {
    loadHighchartsModules(async () => {
      const topology = await fetch(
        "https://code.highcharts.com/mapdata/custom/world.topo.json"
      ).then((response) => response.json());

      setOptions({
        chart: {
          map: topology,
        },
        series: [
          {
            joinBy: ["iso-a3", "code"],
          },
        ],
      });
    });
  }, []);

  if (!options) {
    return <div>Loading...</div>;
  }

  return (
    <HighchartsReact
      highcharts={Highcharts}
      options={options}
      constructorType="mapChart"
    />
  );
}
```

**Comment by Bugrabugra** [comment]
> Hey @Bugrabugra,
> 
> Have you tried loading isolating all your highcharts imports into a single component and loading it with `next/dynamic`? That seems to have fixed the build error for me with highcharts v12.

I have tried dynamic import for "highcharts/modules/map" module at the map component. I haven't tried the way you did, like importing all modules inside a component and calling it with dynamic.

**Comment by Bugrabugra** [comment]
> Hi [Bugrabugra](https://github.com/Bugrabugra)!
> 
> It seems the issue comes from referencing the web API in the node environment during the build. Using the dynamic imports should work as @julioxavierr mentioned, but also, I'd like to refer you to this workaround (to make sure the modules load in the client environment):
> 
> ```
> "use client";
> 
> import { useEffect, useState } from "react";
> import Highcharts from "highcharts";
> import HighchartsReact from "highcharts-react-official";
> 
> const loadHighchartsModules = async (callback) => {
>   Promise.all([
>     import("highcharts/modules/map"),
>     import("highcharts/modules/exporting"),
>     import("highcharts/modules/accessibility"),
>   ]).then(callback);
> };
> 
> export default function Home() {
>   const [options, setOptions] = useState(null);
> 
>   useEffect(() => {
>     loadHighchartsModules(async () => {
>       const topology = await fetch(
>         "https://code.highcharts.com/mapdata/custom/world.topo.json"
>       ).then((response) => response.json());
> 
>       setOptions({
>         chart: {
>           map: topology,
>         },
>         series: [
>           {
>             joinBy: ["iso-a3", "code"],
>           },
>         ],
>       });
>     });
>   }, []);
> 
>   if (!options) {
>     return <div>Loading...</div>;
>   }
> 
>   return (
>     <HighchartsReact
>       highcharts={Highcharts}
>       options={options}
>       constructorType="mapChart"
>     />
>   );
> }
> ```

Thank you so much, I think this approach was the only thing I haven't tried! It worked at both environments, dev and build. 

Thanks you!

**Comment by wesngu28** [comment]
Running into this issue on dev using the recommended method on the client
```
TypeError: e.chart.options.accessibility is undefined
    o accessibility.js:12
    o accessibility.js:12
    j highcharts.js:9
    j highcharts.js:9
    render highcharts.js:9
    redraw highcharts.js:9
    redraw highcharts.js:9
    redraw highcharts.js:9
    update highcharts.js:9
    d React
 

The above error occurred in the <Unknown> component. It was handled by the <ReactDevOverlay> error boundar
y.```

Am i missing something obvious here? It works fine when I build.

**Comment by KamilKubik** [comment]
Hi @wesngu28, seems the accessibility module works fine with the newest HC version, [demo](https://stackblitz.com/edit/stackblitz-starters-l7x63tkh?description=The%20React%20framework%20for%20production&file=app%2Fpage.tsx,app%2Flayout.tsx&title=Next.js%20Starter).

Since the issue appears locally (dev), could you reproduce it? Feel free to overwrite the above demo.

Regards!

**Comment by wesngu28** [comment]
I figured it out, I was trying to type the options object using Highcharts.Options (not sure if this would have been the right one), but evidently accessibility doesn't exist in its definition.

**Comment by stovmascript** [comment]
@KamilKubik We ran into an issue with the offline exporting module when using your async loading method. It tries to use some destructured `ajax` property from the core components of Highcharts. I didn't dig further into the details of why it's a problem.

Since we have Highcharts isolated in a single component, loading our wrapper component with `next/dynamic` as @julioxavierr mentioned solves it for us. There's also a nice bonus of the whole thing becoming and async loaded chunk.

**Comment by KamilKubik** [comment]
Hi [pgeday](https://github.com/pgeday)! 

Even though I cannot reproduce this specific issue, I've managed to run into similar (hopefully) ones. Here are my thoughts and recommendations: 
- Next.js v15 leads me without errors, but switching to the lower version (I've tested with v13) throws me modules importing errors. 
- Regarding the above (but not specifically), I'd still recommend testing loading modules following [this](https://github.com/highcharts/highcharts/issues/22318#issuecomment-2627405014) approach. By doing so, we should have 100% certainty that the modules are loaded client-side (without any unwanted server-side rendering caused by Next.js during the build). 
- You can try switching the modules importing order. Apart from general rules for a few of them, it may be helpful, as mentioned in [this](https://github.com/highcharts/highcharts-react/issues/505#issuecomment-2641005421) thread. 
- The major difference between the dev and prod environments comes from building the static pages. Depending on the Next.js version, it could be crucial to mark client components properly. 

To summarise, the second point should be the starting point for testing, as there have been a few issues with loading modules in the node environment reported recently. Therefore, I'd confirm that everything is loaded properly (on the client side). 

Also, note that from the Highcharts v12, you should import the modules directly, instead of initializing them (see the [Upgrade notes](https://www.highcharts.com/changelog/#highcharts-v12.0.0) section).

Looking forward to your response,
Kind Regards!

**Comment by pgeday** [comment]
Hi @KamilKubik,
Thanks for your quick and detailed responses. I indeed use Next.js 13.0.5 and I do not have capacity to upgrade it to v15 now.
 
I tested what I understood from the links you provided. I also moved the various import up and down. I also tried to console log if there is a HeatMap and whether it is a function or an object. I get a detailed HeatMap object as if it was imported and in my console shows that the HeatMap is an object, not a function. However, if I change the 
`(typeof HeatMap === 'function') && HeatMap(Highcharts);`
line to
`(typeof HeatMap === 'object') && HeatMap(Highcharts);`
then all the charts break and nothing shows up.

In my dynamic import I have ssr: false. The part of the app that has the charts only loads in after login and nothing is a static page.

Sorry, but what does it mean that you should import the modules directly instead of initializing them. I am importing all the modules per ES6. In the upgrade notes link you shared there is one mention of "import" at v5.0.0. Initialize has 1 mention at v7.1.3, unrelated. 

I paid for the Advantage support is there a way of a direct look at the code and discuss?

Thanks,
Peter

**Comment by KamilKubik** [comment]
Thanks for your clarification, @pgeday. 

I've managed to reproduce this issue by using a live environment with files uploaded to the server. It seems to be module-related and, more specifically, to how Next.js treats rendering. Highcharts should be rendered on the client side, and, in theory, we shouldn't encounter any errors when running it in an 'isolated' client-side environment, which we are trying to achieve using the suggestions above. 

It seems using the ESM imported in an 'isolated' way along with the Next.js [v13.0.7](https://github.com/vercel/next.js/releases/tag/v13.0.7) fixes all errors in my environment. I'm sharing the example code below (you should be able to render this component without dynamic import). Could you bump up your version and confirm whether it works for you as well? I can see that upgrading to the latest version is not an option, but hopefully, upgrading to version 13.0.7 shouldn't affect the rest of your application. Also, to answer your question regarding the initialization, the code you shared is the correct approach (importing ESM, also shown below). 
```
"use client";

import React, { useEffect, useState } from "react";
import Highcharts from "highcharts/es-modules/masters/highcharts.src.js";
import HighchartsReact from "highcharts-react-official";

const loadHighchartsModules = async (callback: () => void) => {
  Promise.all([
    import("highcharts/es-modules/masters/modules/histogram-bellcurve.src.js"),
    import("highcharts/es-modules/masters/modules/draggable-points.src.js"),
    import("highcharts/es-modules/masters/modules/sankey.src.js"),
    import("highcharts/es-modules/masters/modules/exporting.src.js"),
    import("highcharts/es-modules/masters/highcharts-more.src.js"),
    import("highcharts/es-modules/masters/modules/heatmap.src.js"),
    import("highcharts/es-modules/masters/modules/accessibility.src.js"),
  ]).then(callback);
};

const HeatmapChart: React.FC = () => {
  const [options, setOptions] = useState({
    isLoading: true,
    data: {
      chart: {
        type: "heatmap",
        marginTop: 40,
        marginBottom: 80,
        plotBorderWidth: 1,
      },
      title: { text: "Title information", style: { fontSize: "1em" } },
      xAxis: { categories: ["S 1", "S 2"] },
      yAxis: { categories: ["T 1", "T 2"], title: null, reversed: true },
      accessibility: {
        point: {
          descriptionFormat:
            "{(add index 1)}. {series.xAxis.categories.(x)} {series.yAxis.categories.(y)}, {value}.",
        },
      },
      colorAxis: { minColor: "#00e272", maxColor: "#fa4b42" },
      legend: {
        align: "right",
        layout: "vertical",
        margin: 0,
        verticalAlign: "top",
        y: 25,
        symbolHeight: 280,
      },
      tooltip: {
        format:
          "<b>{point.value}</b> from <br><b>{series.yAxis.categories.(point.y)}</b> at<br><b>{series.xAxis.categories.(point.x)}</b>",
      },
      credits: { enabled: false },
      series: [
        {
          borderWidth: 1,
          data: [
            [0, 0, 904.75],
            [1, 0, 904.75],
            [0, 1, 904.75],
            [1, 1, 904.75],
          ],
          dataLabels: {
            enabled: false,
            color: "#000000",
          },
        },
      ],
      responsive: {
        rules: [
          {
            condition: { maxWidth: 500 },
            chartOptions: {
              yAxis: { labels: { format: "{substr value 0 1}" } },
            },
          },
        ],
      },
    },
  });

  useEffect(() => {
    loadHighchartsModules(() => {
      setOptions((prev) => ({ ...prev, isLoading: false }));
    });
  }, []);

  if (options.isLoading) {
    return <div>Loading...</div>;
  }

  return <HighchartsReact highcharts={Highcharts} options={options.data} />;
};

export default HeatmapChart;
```

Advantage support is provided through an internal support channel, and I believe that I or my colleague will guide you to make it work. As the code is now reproducible, this seems sufficient for debugging, but we are still somewhat dependent on the Next.js specific environment. Let me know if you found the above helpful. 

Regards!

**Comment by nimbit-software** [comment]
HI this sort of works but it seems to be a bit of a workaround. Is this the official way to hanlde this or is there something in the works? Thanks in advance

**Comment by KamilKubik** [comment]
Hi [nimbit-software](https://github.com/nimbit-software), we are actively working on an official, stable solution. As you mentioned, I would treat the approaches discussed here as workarounds. Apart from the official docs, I will leave a reference here as soon as it is fixed. 

Regards!

**Comment by nimbit-software** [comment]
Awesome we will wait with the migration then. If there is anything we can do to help let me know.

**Comment by jaredterm** [comment]
Hey @KamilKubik just FYI, I run into the same build issue in Next 15.2.4 when trying to update from Highcharts 11.4.8 to latest (currently 12.1.2)

I am also using ESM to import Highcharts.

```shell
pnpx envinfo --system --npmPackages highcharts,next,react --binaries --browsers

  System:
    OS: macOS 15.3.2
    CPU: (8) arm64 Apple M1 Pro
    Memory: 745.75 MB / 16.00 GB
    Shell: 5.2.37 - /opt/homebrew/bin/bash
  Binaries:
    Node: 22.14.0 - ~/.nave/installed/22.14.0/bin/node
    npm: 10.9.2 - ~/.nave/installed/22.14.0/bin/npm
    pnpm: 10.7.0 - ~/.nave/installed/22.14.0/bin/pnpm
  Browsers:
    Safari: 18.3.1
  npmPackages:
    highcharts: ^12.1.2 => 12.1.2 
    next: 15.2.4 => 15.2.4 
    react: 19.1.0 => 19.1.0 
```

**Comment by KamilKubik** [comment]
@jaredterm, thanks for reaching out.

Have you tried using the solution mentioned [here](https://github.com/highcharts/highcharts/issues/22318#issuecomment-2627405014)? 

Regards!

**Comment by nimbit-software** [comment]
@KamilKubik Is there any update on this? or can we get support on this we have a Highcharts licence with support?

**Comment by KamilKubik** [comment]
[nimbit-software](https://github.com/nimbit-software) Yes, and thanks for asking!

We've recently introduced the ESM bundles (Highcharts v12.2.0) as an official solution that should fully support the client/node environments. 

Here's the related [comment](https://github.com/highcharts/highcharts-react/issues/505#issuecomment-2865695253), and the updated imports based on the example shared above:

```
import Highcharts from "highcharts/esm/highcharts.js";
import HighchartsReact from "highcharts-react-official";

import "highcharts/esm/highcharts-more.js";
import "highcharts/esm/modules/heatmap.js";
import "highcharts/esm/modules/histogram-bellcurve.js";
import "highcharts/esm/modules/draggable-points.js";
import "highcharts/esm/modules/sankey.js";
import "highcharts/esm/modules/exporting.js";
import "highcharts/esm/modules/accessibility.js";
```

[Here](https://www.highcharts.com/docs/getting-started/installation-with-esm#dynamic-imports-from-cdn) is a general docs mention, but we also have an open [issue](https://github.com/highcharts/highcharts-react/issues/518) regarding the more React/Next.js based docs/examples that we're working on. 

I'm happy to help, but if you'd like to reach out via the Highcharts Advantage channels, feel free to do so under [those](https://www.highcharts.com/support/#advantage-support-section) sections. 

Looking forward to your response on how this works in your case.
Kind Regards!

**Comment by gareth-m95** [comment]
Hi @KamilKubik when import the accessibility module via esm `import "highcharts/esm/modules/accessibility.js";`, it doesn't actually appear to import the accessibility module - the features stop working and the 'Consider including the "accessibility.js" module' warning gets triggered

**Comment by oysteinmoseng** [comment]
Should be closed by https://github.com/highcharts/highcharts/pull/23090 now that we get proper docs for NextJS. Please comment to reopen if anything remains unclear or is not working as expected!

**Comment by julioxavierr** [comment]
@oysteinmoseng the doc appears to focus on Next.js with the App Router. However, I believe some users still utilize the Pages Router, so it is worth documenting. In this case, the solution could involve using a dynamic import.
